example/src/examples/http_proxy_repository.dart

```diff
@@ -29,16 +29,24 @@ class HttpProxyRepository extends PackageRepository {
           baseUrl.resolve('/api/packages/${Uri.encodeComponent(package)}');
 
       http.Response response = await client.get(versionUrl);
```

Comment:
Thanks for the pull request.

Maybe you could use `response.statusCode` instead here? I think it should return 200 if everything went fine and 404 if it was not found (see the protocol description [here](https://github.com/dart-lang/pub_server/blob/master/lib/shelf_pubserver.dart#L31) )?